### ISSUE TITLE
Reparenting Light, Camera and CSGBox cause error Condition "!id_map.has(p_rid.get_data())" is true. Returned: __null

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#45461]
**Godot version:**
3.2.4.beta.custom_build. 2d904127d

**Issue description:**
When running project then this error is shown:
```
Condition "!id_map.has(p_rid.get_data())" is true. Returned: __null
```
Scene contains 3 types of nodes:
- Camera
- CSGBox
- OmniLight

which are randomly reparented, deleted or added.

![Zrzut ekranu z 2021-01-25 22-11-04](https://user-images.githubusercontent.com/41945903/105766591-3d980380-5f5a-11eb-9dfc-5cff069dfbc7.png)

**Steps to reproduce:**
1. Run project

**Minimal reproduction project:**

[RegressionTestProject.zip](https://github.com/godotengine/godot/files/5869687/RegressionTestProject.zip)

**Comment by lawnjelly** [comment]
This is a BVH bug, it disappears when I switch off BVH (`project_settings->rendering->quality->spatial_partitioning->use_bvh`). I'll take a look soon (may be busy part of tomorrow).

**Comment by lawnjelly** [comment]
This has been incredibly difficult to debug due to the random nature, but I think I have it figured out.

I think it is sending the unpairing callback twice. I didn't think this should happen but must be a bug, hopefully will be much easier to solve now I know where it is going wrong.

**Comment by akien-mga** [comment]
Fixed by #45507.

**Comment by qarmin** [comment]
Maybe I'm doing something wrong, but I still can reproduce errors..

**Comment by lawnjelly** [comment]
Ah you are right, I'd modified your project slightly, and it had fixed that but not with the value set to 19. I'll have another look it may be a similar fix for this.

**Comment by akien-mga** [comment]
Fixed by #45530 (may need another test to double check).
